### PR TITLE
Upgraded dependencies; adjusted not to define resolve() as undefined

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   },
   "scripts": {
     "build": "rimraf lib && tsc",
-    "lint": "tslint \"src/**/*.ts\" --type-check --project ./tsconfig.json",
-    "lint:fix": "tslint --fix \"src/**/*.ts\" --type-check --project ./tsconfig.json",
+    "lint": "tslint \"src/**/*.ts\" --project ./tsconfig.json",
```
Comment (tibawatanabe) [inline package.json:11]: Hum... why did you remove the type-check?

File: package.json [hunk 2]
```diff
@@ -26,21 +26,21 @@
...
-    "graphql": "0.11.7",
-    "mocha": "2.5.3",
-    "assert": "1.4.1",
-    "rimraf": "2.5.2",
-    "tslint": "5.1.0",
-    "typescript": "2.3.4",
-    "@types/graphql": "0.11.5",
-    "@types/mocha": "2.2.39",
-    "@types/node": "7.0.5"
+    "@types/graphql": "^0.13.1",
```
Comment (tibawatanabe) [inline package.json:35]: We fix the version here to avoid problems with unintended updated

File: src/type-factory/field.type-factory.ts [hunk 3]
```diff
@@ -224,7 +232,7 @@ export function fieldTypeFactory(target: Function, metadata?: FieldMetadata, isI
     type: fieldType,
     description: metadata.description,
     args: args,
-    resolve: resolveFn,
+    ...(resolveFn ? { resolve: resolveFn } : {}),
```
Comment (tibawatanabe) [inline src/type-factory/field.type-factory.ts:235]: Hum... could you explain why using this approach is better?

Comment (alexey-pelykh) [reply src/type-factory/field.type-factory.ts:235]: Because with recent changes in `graphql` itself the `resolve` field being set results an error.

File: package.json (hunk 2 above)
Comment (alexey-pelykh) [reply package.json:35]: I guess then it should be fixed to `0.13.1`

File: package.json (hunk 1 above)
Comment (alexey-pelykh) [reply package.json:11]: Probably this can be safely reverted

File: src/type-factory/field.type-factory.ts (hunk 3 above)
Comment (MichalLytek) [reply src/type-factory/field.type-factory.ts:235]: spread operator with object is just a hack, not a solution for type changes

Comment (alexey-pelykh) [reply src/type-factory/field.type-factory.ts:235]: Agree, yet there's still an issue with `resolve`
